### ISSUE TITLE
Add error codes to all errors

### ISSUE THREAD

**Issue by tonybaloney** [tonybaloney/wily#16]
in the form {module-#}  - {ref}
add the module # to the docstring at the top of each module.

**Comment by Kilo59** [comment]
Do you have any preference for the format of the module number within the docstring?
```python
"""
blah blah blah
Module#7
"""
```

What would the `{ref}`erence actually reference?

----------------------------

Lastly have you looked into the [better-exceptions](https://github.com/Qix-/better-exceptions) packages? When it works it's great but I've also run into a couple of wierd edge cases that seemed to break it.

**Comment by Kilo59** [comment]
Cool, I'm happy to take this.
Should be able to get to it before next week.

**Comment by DahlitzFlorian** [comment]
Hey @Kilo59! Are you still working on this and if so, are you making progress or do you need help?

**Comment by Kilo59** [comment]
Yeah, I am, but I totally forgot about it, thanks for the reminder. I just opened an initial PR with my idea for Module numbering to get some feedback before I start trying to add the module numbers to the error codes.
